models/repo.go

```diff
@@ -1023,24 +1023,30 @@ func CheckCreateRepository(doer, u *User, name string) error {
 	} else if has {
```

Comment:
I think we have checked if it exists here.

Reply:
Not in the same way.
isRepositoryExist requires that there is a dB entry and the repo files exists.

The next test just checks if there are files present.